### PR TITLE
Support sreh in copy from parse stage

### PR DESCRIPTION
**Pull request by weinan003**
After commit f82de5c46bd, COPY FROM add a parse stage at the very begin process. Our sreh can not handle the error which the data have encoding error.

Since the raw_buf,input_buf and line_buf does not cleanup after parse stage error have raised, even sreh cache the ERROR message, copy infinitly stuck at the error stage.

Fix:
if the parse error, in the stage internal, skip the error record by EOL.

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by my-ship-it** [comment]
Please fix CI failures?

**Comment by weinan003** [comment]
> Please fix CI failures?

Compared with the upstream's answer file, our answer file is this bug we have fixed. update the correct answer file back

File: src/backend/commands/copyfromparse.c [hunk 1]
```diff
@@ -1735,3 +1785,92 @@ CopyReadBinaryAttribute(CopyFromState cstate, FmgrInfo *flinfo,
...
+	EolType eol_type = cstate->opts.eol_type;
+	bool found = false;
+	for (*scanidx = 0;
+		 *scanidx < (cstate->raw_buf_len - cstate->raw_buf_index);
+		 (*scanidx)++)
+	{
+		char c1 = cstate->raw_buf[cstate->raw_buf_index + *scanidx];
+		char c2 = cstate->raw_buf[cstate->raw_buf_index + *scanidx + 1];
+		if (((eol_type == EOL_CRNL || eol_type == EOL_UNKNOWN) && c1 == '\r' &&
+			 c2 == '\n'))
```
Comment (avamingli) [inline src/backend/commands/copyfromparse.c:1860]: Is it possible index `*scanidx + 1` beyond the bound of raw_buf?

Comment (weinan003) [reply src/backend/commands/copyfromparse.c:1860]: nice cache!

**Review by avamingli — APPROVED** [review]

**Pull request merged**